src/lib/mathlib/math/filter/AlphaFilter.hpp

```diff
@@ -70,6 +73,21 @@ class AlphaFilter
 		}
 	}
 
+	void setCutoffFreq(float sample_freq, float cutoff_freq)
+	{
+		if ((sample_freq <= 0.f) || (cutoff_freq <= 0.f) || (cutoff_freq >= sample_freq / 2.f)
+		    || !isFinite(sample_freq) || !isFinite(cutoff_freq)) {
+
+			// No filtering
+			_alpha = 1.f;
```

Comment:
Sorry for seeing this late but I would have put alpha to zero in case the sanity checks don't pass because that avoids any situation where the caller assumes the values get filtered but instead it just passes them back.

As discussed this would not make the filter pass through with the cutoff set to zero, that could be separate.
I don't feel very strong about this, just a suggestion.

Reply:
@MaEtUgR Should that function then return a boolean to tell if the settings were properly applied?

Reply:
> @MaEtUgR Should that function then return a boolean to tell if the settings were properly applied?

That sounds like a good pattern I wouldn't mind replicating in the biquads.

Reply:
> Should that function then return a boolean to tell if the settings were properly applied?

True, that would be an even clearer interface. The function would not change the parameters if it fails the checks but return false. So alpha would just stay zero if it was not initialized before.